openapi_python_client/__init__.py

```diff
@@ -239,15 +239,30 @@ def _build_api(self) -> None:
         client_path.write_text(client_template.render(), encoding=self.file_encoding)
 
         # Generate endpoints
+        endpoint_collections_by_tag = self.openapi.endpoint_collections_by_tag.items()
         api_dir = self.package_dir / "api"
         api_dir.mkdir()
-        api_init = api_dir / "__init__.py"
-        api_init.write_text('""" Contains methods for accessing the API """', encoding=self.file_encoding)
+        api_init_path = api_dir / "__init__.py"
+        api_init_template = self.env.get_template("api_init.py.jinja")
+        api_init_path.write_text(
+            api_init_template.render(
+                package_name=self.package_name,
+                endpoint_collections_by_tag=endpoint_collections_by_tag,
```

Comment:
I assume you're passing these params because you need them, but for a developer working on just `openapi-python-client` it looks like these are not needed and might get removed at some point. I suggest either adding a note here or, better yet, a custom template in the end to end tests which uses them so we can see why they're there.

Reply:
Good catch. I followed your recommendation:
- https://github.com/triaxtec/openapi-python-client/pull/403/commits/eda38edc630c3e65116b65963739abe7e15e2acb Add new custom template tests, I had to change a little bit how the expected diff was handled:
  - Instead of a dict of {filename: content} it now use a dict of {file path: content}, to be able to handle files that have the same name but at are defined at different paths
  - Since I've added a custom template sample, directly hardcoding the expected content seems to become a bad idea. That's why I propose to introduce the `custom-templates-golden-record` which contains all expected diff when using the custom template
  
  https://github.com/triaxtec/openapi-python-client/pull/403/commits/96fcdd6d42746711e9d1fa6f8f30c028c2d709b8 add the `custom-templates-golden-record`  generation to the `regen` task

Reply:
We used to generate a complete golden-record for custom to compare as well but code reviews got really burdensome since the same content would effectively change twice and it didn't add much value. At a glance, it looks like you've set it up to only generate the specific files we're interested in so that should be a good middleground between the two approaches (generate everything vs generate nothing). I'll take a deeper look later on, thanks!